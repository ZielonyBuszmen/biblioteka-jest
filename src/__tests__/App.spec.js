import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import App from '../App';

configure({adapter: new Adapter() });

describe("App", () => {
    it("should render single image",  () => {
        const wrapper = shallow(<App />);
        wrapper.setState({
            imageId: 0,
            imageList: ['firtst', 'second'],
        });

        expect(wrapper).toMatchSnapshot();
    });

    it("should render gallery", () => {
        const wrapper = shallow(<App />);
        wrapper.setState({
            imageId: null,
        });

        expect(wrapper).toMatchSnapshot();
    });
});