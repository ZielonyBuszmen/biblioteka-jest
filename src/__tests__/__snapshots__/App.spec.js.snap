// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render gallery 1`] = `
<div>
  <Header
    gotoMainSiteHandler={[Function]}
    isShowImage={false}
  />
  <div
    className="container"
  >
    <Gallery
      gotoImageHandler={[Function]}
      imageList={null}
    />
  </div>
</div>
`;

exports[`App should render single image 1`] = `
<div>
  <Header
    gotoMainSiteHandler={[Function]}
    isShowImage={true}
  />
  <div
    className="container"
  >
    <Image
      addCommentHandler={[Function]}
      addLikeHandler={[Function]}
      image="firtst"
      imageId={0}
    />
  </div>
</div>
`;
